pkg/ebpf/tracee.go

```diff
@@ -162,10 +162,16 @@ type fileExecInfo struct {
 	Hash      string
 }
 
+type handledEvent struct {
+	load   bool // event's bpf programs should be loaded into the kernel
+	submit bool // event should be submitted to userspace
```

Comment:
One example of such programs are the sys_enter/sys_exit events, which are used by any syscall event, but can also be sent as an event on its own. Then, you want to load the sys_enter/sys_exit probes, but not to send these events if they were not chosen.
Another example is the sched_process_fork program, which should be loaded for the proper functionality of tracee (updating relevant maps), but will not always be chosen as an event.
And there is the example you gave of the magic_write event.Where the same bpf program is used for different events - vfs_write/vfs_writev/magic_write.
So the idea is that we load programs required by the events which were chosen, or required for the proper functionality of tracee, however, not always we want to send all the events which can be sent from these programs

Reply:
Thanks for the explanation.
Maybe not in the scope of this PR, but WDYT about creating a map, similar to the events definitions, of probes? This way we can reuse those probes in the event's definitions probe dependencies. This would provide an explicit interface to the eBPF programs we write for tracee, instead of fusing them with the events (which may not necessarily correspond to a program).

Reply:
Yes, we can add such a map of probes, and it might simplify things.
Let's consider this in a future PR